### ISSUE TITLE
deepin-editor root

### ISSUE THREAD

**Issue by antechdesigns** [linuxdeepin/developer-center#889]
## Describe the bug
deepin-editor does not open as root from command line

## To Reproduce in terminal:
sudo deepin-editor /path/to/file.conf

##output error:
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
"dtkwidget2" can not find qm files
"deepin-editor" can not find qm files

## Expected behavior
Open and edit file as root

### Distro & Version
archlinux deepin-editor 1.2.2

### Additional context
deepin-editor /path/to/file.conf does work (no errors)

**Comment by aiamuzz** [comment]
seems like an arch specific bug ... opens just fine in DeepinOS ...

@antechdesigns ... FYI ... we can use `dedit` instead of typing the whole `deepin-editor` in the terminal !!!

**Comment by antechdesigns** [comment]
> seems like an arch specific bug ... opens just fine in DeepinOS ...
> 
> @antechdesigns ... FYI ... we can use `dedit` instead of typing the whole `deepin-editor` in the terminal !!!

thanks for the info @aiamuzz Yep looks like a arch specific bug

**Comment by keybreak** [comment]
Yes confirm, this is Arch only, i also see this in Manjaro Deepin after latest 10.02.2019 update, before it was ok.

**Comment by BLumia** [comment]
@listenerri can you reproduce the bug?

**Comment by antechdesigns** [comment]
sudo dedit /etc/default/grub
`QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'`

**Comment by keybreak** [comment]
Also this error occurs if you'll try to open anything with Deepin editor under root file manager

**Comment by antechdesigns** [comment]
> Also this error occurs if you'll try to open anything with Deepin editor under root file manager...
> Which pretty much results in one simple fact - now you can't use deepin-editor as root at all )

Yep, I have to install gedit to work around this problem.

**Comment by keybreak** [comment]
@felixonmars 
Please check it, surely you can confirm this one in Arch)

**Comment by antechdesigns** [comment]
@BLumia @felixonmars 

Still not fixed, thanks ;)

**Comment by BLumia** [comment]
Confirmed, will look into it later :)

**Comment by antechdesigns** [comment]
@BLumia 

Any chance of this being fixed before April (this year) ;)
I want to make a new archdeepin.ISO without having to add extra pkgs e.g gedit

**Comment by BLumia** [comment]
> @BLumia
> 
> Any chance of this being fixed before April (this year) ;)
> I want to make a new archdeepin.ISO without having to add extra pkgs e.g gedit

Will definitely look into this issue when I have spare time. Sorry my schedule is quite full in this month :(

Anyone would like to help are welcome

**Comment by gobonja** [comment]
This issue occurs due to dbus-1.12.12 and 1.12.10-**2** .

Solutions: 
- run with dbus-launch (e.g. `sudo dbus-launch dedit`)
- or install dbus-x11 from AUR
- or downgrade to dbus-1.12.10-**1**

It looks like  [X11 autolaunching](https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/dbus&id=d915d6c247a447bdcf710b76aac085e7cd189075) is disabled from version 1.12.10-**2**

Deepin uses version dbus-1.12.8. If you manually upgrade to version 1.12.12  from testing repos same issue occurs.

**Comment by keybreak** [comment]
@BLumia i confirm `sudo dbus-launch dedit` works, @gobonja got point!
Please fix it :smiley:

**Comment by antechdesigns** [comment]
installing gedit it is then...

**Comment by keybreak** [comment]
@BLumia 
Please reopen this and solve, so it wouldn't get lost.

**Comment by BLumia** [comment]
> @BLumia
> Please reopen this and solve, so it wouldn't get lost.

Here you go, and hope anyone familiar with d-bus policy could give us a hand in case I could mess it up :joy:

**Comment by keybreak** [comment]
Btw it's not only commandline it doesn't open as root also from deepin-file-manager and everywhere else :stuck_out_tongue_winking_eye:
